src/forms/inputcounter/InputCounter.tsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+import * as React from "react";
+import * as utils from "../../utils";
+import {Icon} from "../../icon";
+import {Input} from "../input/Input";
+
+type InputCounterProps = {
+    className?: string;
+    maxValue?: number;
+    minValue?: number;
+    defaultValue?: number;
+    name?: string;
+    errMsg?: string;
+    width?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+type InputCounterState = {
+    value: number;
+    errorMsg: string;
```

Comment:
Instate of keeping error message in state can we derive it at runtime ?

Reply:
what is the problem in state? If we remove it from state, then we need to use ref that add complexity in code.

Reply:
There is no problem here. But  just a thought how does error message will affect state of this component? .Does this is something that component want to remember throughout state change and  does next state depends on error message?